docs/backend/guides/how-to/create-a-custom-linter.md

```diff
@@ -0,0 +1,137 @@
+
+```golang
+func run(pass *analysis.Pass) (interface{}, error) {
+    // Some code
+}
+```
+
+The linter is actually analyzing an abstract syntax tree, AST, that represents code in a file. 
+When your linter gets to a position in a file, where it catches an error, bug, or issue, it will flag this for the user.
+Because the linter is analyzing an AST, your code must be able to search through a file and mark which position it is when an issue is caught.
+
+A great resource to use to learn how to write code for analyzing an AST can be found [here](https://disaev.me/p/writing-useful-go-analysis-linter/).
+If you're curious about what an AST looks like for your program, checkout [this](http://goast.yuroyoro.net/) online visualizer.
+
+### Writing linter tests
+Linter tests also do not look liek your typical go tests. They function with want statements.
+
+You will still create tests that test the happy and unhappy paths of your code. However, rather than your typical expect
+statement, you will instead put a want statement as a comment next to where you expect your linter to flag an issue:
+
+```golang
+package examplelinter
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"golang.org/x/tools/go/analysis/analysistest"
+)
+
+// this test starts up Test runner and looks at tests in example_linter_tests and runs linter against those files
+// if there are no want statements, the linter moves on to the next statement
+// if there are no want statements and the test fails, our linter is failing because nothing is expected.
+func TestAll(t *testing.T) {
+	wd, err := os.Getwd()
+	if err != nil {
+		t.Errorf("Failed to get wd: %s", err)
+	}
+
+	testdata := filepath.Join(filepath.Dir(filepath.Dir(wd)), "testdata")
+
+	// Pass in the linter that we want to use, and location of linter tests:
+	analysistest.Run(t, testdata, LinterAnalyzer, "example_linter_tests/...")
+}
+```
+
+## Testing the linter across files
+
+To run your linter across files run this command in your terminal:
+`go run ./cmd/example-linter/main.go -- ./...`
+
+The `-- ./...` flag tells the linter to run against __all__ files.
+
+To run the linter tests only:
+`go test ./pkg/example-linter/... -v`
+
+## Running the linter in pre-commit
+Depending on the function of your linter, you may want to add it to precommit. You will do this by adding a new definition to precommit:
+```golang
+  - repo: local
+    hooks:
+      - id: example-linter
+        name: example-linter
+        entry: scripts/pre-commit-go-custom-linter example-linter
+        files: \.go$
+        pass_filenames: false
+        language: script
+```
+
+Note that the `language` key refers to a bash script, that is because we have had the most success with 
+creating a custom bash script (located in the `script` folder) for running all custom linters in precommit.
+
+You can checkout this custom bash script at `scripts/pre-commit-go-custom-linter`.
+
+### Additional Resources:
```

Comment:
I'm not sure if it should go here or scattered throughout the docs, but it might be helpful to point the docs to one or both of the existing linters to show what they look like in `cmd`, `pkg`, and `testdata`.